src/body.rs

```diff
@@ -612,6 +619,21 @@ impl LineMetaMut for &mut LineBuilder {
     }
 }
 
+impl LineBufferMut for &mut LineBuilder {
+    fn get_line_buffer(&mut self) -> Option<&[u8]> {
+        self.line.as_deref().map(|x| x.as_bytes())
+    }
+    fn set_line_buffer(&mut self, line: Vec<u8>) -> Result<(), LineMetaError> {
+        self.line = Some(
+            from_utf8(&line)
```

Comment:
If we use http://doc.rust-lang.org/std/string/struct.String.html#method.from_utf8 we can convert directly to String and reuse the Vec allocation rather than converting to str then String

Reply:
oh, TIL, will change!